test/dependency-manager/volumes.test.ts

```diff
@@ -0,0 +1,89 @@
+import { expect } from '@oclif/test';
+import axios from 'axios';
+import mock_fs from 'mock-fs';
+import sinon from 'sinon';
+import Build from '../../src/commands/build';
+import LocalDependencyManager from '../../src/common/dependency-manager/local-manager';
+import * as DockerCompose from '../../src/common/docker-compose';
+import DockerComposeTemplate from '../../src/common/docker-compose/template';
+
+describe('volumes', function () {
+  let compose: DockerComposeTemplate;
+
+  beforeEach(async () => {
+    // Stub the logger
+    sinon.replace(Build.prototype, 'log', sinon.stub());
+
+    const backend_json = {
+      "name": "architect/backend",
+      "parameters": {
+        "VOLUME_PATH": {
+          "default": '/tmp/volume_path'
+        }
+      },
+      "volumes": {
+        "service_config_volume": {
+          "mountPath": "/tmp/volume"
+        },
+        "volume_from_param": {
+          "mountPath": "$VOLUME_PATH"
+        },
+        "split_volume": {
+          "mountPath": "/tmp:/vol/service_config"
+        },
+        "service_config_override": {
+          "mountPath": "/tmp/overridden_path"
+        }
+      }
+    };
+
+    const env_config = {
+      "services": {
+        "architect/backend:latest": {
+          "debug": {
+            "path": "./src/backend",
+            "volumes": {
+              "service_config_override": "/tmp/service_config_overridden",
+              "env_defined_volume": "/tmp/env_defined_volume"
+            }
+          }
+        }
+      }
+    };
+
+    mock_fs({
+      '/stack/src/backend/architect.json': JSON.stringify(backend_json),
+      '/stack/arc.env.json': JSON.stringify(env_config),
+    });
+
+    const manager = await LocalDependencyManager.createFromPath(axios.create(), '/stack/arc.env.json');
+    compose = DockerCompose.generate(manager);
+  });
+
+  afterEach(function () {
+    // Restore stubs
+    sinon.restore();
+    // Restore fs
+    mock_fs.restore();
+  });
+
+  it('service config volume', async () => {
+    expect(compose.services['architect.backend.latest'].volumes).to.include.members(['/tmp/volume']);
```

Comment:
These tests should be more specific and test that we're successfully joining the host path from the environment config with the container path from the service config.

Reply:
I think I see what you're saying now actually. I misunderstood how you wanted the env config values to map to service config values